frontend/src/components/resourceMap/useQueryParamsState.tsx

```diff
@@ -18,64 +21,43 @@ export function useQueryParamsState<T extends string | undefined>(
   param: string,
   initialState: T
 ): UseQueryParamsStateReturnType<T> {
-  const location = useLocation();
+  const { search } = useLocation();
   const history = useHistory();
 
-  // State for managing the value derived from the query parameter
-  const [value, setValue] = useState<T | undefined>(() => {
-    const { search } = location;
-    const searchParams = new URLSearchParams(search);
-    const paramValue = searchParams.get(param);
+  const params = new URLSearchParams(search);
+  const value = (params.get(param) ?? undefined) as T | undefined;
```

Comment:
We should memo the value, since this logic should only run if the search changes.

Reply:
done, although this is not a very expensive operation